File: examples/src/Text/GetTextStyleEffectiveData.cs
import aspose.slides as slides
using Aspose.slides.Examples.CSharp
using System
using System.Collections.Generic
using System.Linq
using System.text
using System.Threading.Tasks

namespace CSharp.text
{
    class GetTextStyleEffectiveData
    {
        public static void Run() {

            #ExStart:
            # The path to the documents directory.
            dataDir = RunExamples.GetDataDir_Text()


            using (Presentation pres = new Presentation(dataDir + "Presentation1.pptx"))
            {
                shape = pres.slides[0].shapes[0] as IAutoShape

                ITextStyleEffectiveData effectiveTextStyle = shape.text_frame.TextFrameFormat.TextStyle.GetEffective()

                for (i = 0 i <= 8 i++)
                {
                    IParagraphFormatEffectiveData effectiveStyleLevel = effectiveTextStyle.GetLevel(i)
                    print("= Effective paragraph formatting for style level #" + i + " =")

                    print("Depth: " + effectiveStyleLevel.Depth)
                    print("Indent: " + effectiveStyleLevel.Indent)
                    print("Alignment: " + effectiveStyleLevel.Alignment)
                    print("Font alignment: " + effectiveStyleLevel.FontAlignment)
                }

            }

            #ExEnd:
        }

    }
}

File: examples/src/Text/get_text_style_effective_data.py
import aspose.slides as slides


def get_text_style_effective_data(global_opts):
    with slides.Presentation(global_opts.data_dir + "text_add_animation_effect.pptx") as pres:
        shape = pres.slides[0].shapes[0]
        effective_text_style = shape.text_frame.text_frame_format.text_style.get_effective()

        for i in range(9):
            effective_style_level = effective_text_style.get_level(i)
            print("= Effective paragraph formatting for style level #" + str(i) + " =")

            print("depth: " + str(effective_style_level.depth))
            print("Indent: " + str(effective_style_level.indent))
            print("Alignment: " + str(effective_style_level.alignment))
            print("Font alignment: " + str(effective_style_level.font_alignment))
